spring-core-test/src/main/java/org/springframework/aot/test/agent/EnabledIfRuntimeHintsAgent.java

```diff
@@ -51,13 +50,13 @@
  * </pre>
  *
  * @author Brian Clozel
+ * @author Sam Brannen
  * @since 6.0
  */
 @Target({ElementType.TYPE, ElementType.METHOD})
 @Retention(RetentionPolicy.RUNTIME)
 @Documented
-@ExtendWith(RuntimeHintsAgentCondition.class)
-@Tag("RuntimeHintsTests")
```

Comment:
Why remove the tag here?

Reply:
> Sorry I missed that bit; yes, this tag is used in [our test suite configuration](https://github.com/spring-projects/spring-framework/blob/db43425bf92431d8fb4c73c4b5de3c3f75db47e8/buildSrc/src/main/java/org/springframework/build/hint/RuntimeHintsAgentPlugin.java#L44): we want to run those tests in a separate task, but avoid re-running all other tests.

OK. That makes sense then.

> This annotation is opinionated and the condition implementation was meant to be reused by other Spring projects, with a different setup. 

The annotation remains with the same semantics and effective behavior -- or at least that was my intention.

How would other Spring projects use the custom condition with a different setup?

> I see that the condition has been made package private in [279f55f](https://github.com/spring-projects/spring-framework/commit/279f55fdfa1659897086331cb960f31c627affe3)

Yes, that's the idiomatic way to implement custom conditions that are transparently configured for the user via a custom `public` annotation like `@EnabledIfRuntimeHintsAgent`.

It's rarely the case that users need to access the custom condition class itself, but if you have a use case for that please let me know.

> (and yet marked as `@since 6.0`?). 

It was introduced in 6.0, so that makes sense. `@since` tags are not only for public APIs.

> Should other Spring projects copy our annotation definition in their own project if they want a different setup? Is there a better way to achieve this?

I'm not quite sure what you're asking, but perhaps my previous comments already answer that. If not, please expound.